src/horizontalaligner.cpp

```diff
@@ -461,17 +461,22 @@ void Alignment::ClearGraceAligners()
     m_graceAligners.clear();
 }
 
-std::pair<int, bool> Alignment::GetAccidAdjustedPosition(Doc *doc, int staffN)
+std::pair<int, bool> Alignment::GetAccidAdjustedPosition(Doc *doc, int staffN, int staffSize)
 {
     if (!HasAlignmentReference(staffN)) return { 0, false };
 
     AlignmentReference *reference = GetAlignmentReference(staffN);
     if (!reference) return { 0, false };
 
-    Accid *accid = reference->GetFrontAccid();
-    if (!accid) return { 0, false };
+    const std::vector<Accid *> *alignmentAccidentals = reference->GetAccidSpace();
+    if (alignmentAccidentals->empty()) return { 0, false };
 
-    return { accid->GetDrawingX(), true };
+    auto iter = std::min_element(alignmentAccidentals->cbegin(), alignmentAccidentals->cend(),
+        [](const Accid *left, const Accid *right) { return left->GetDrawingX() < right->GetDrawingX(); });
+    Accid *accid = vrv_cast<Accid *>(*iter);
+    const int accidGlyphWidth = doc->GetGlyphWidth(accid->GetAccidGlyph(accid->GetAccid()), staffSize, false);
```

Comment:
I think you call `Accid->GetSelfLeft()` that will give you the exact x position you are looking for. You want to do the same in the `std::min_element` call above because the one that has the smallest drawing x position is not necessary the one with the smallest self x (depending on their accid glyph). However, calling `GetSelfLeft` would work only if this is not done in the drawing loop but in a separate alignment step. See my general remark.

Reply:
Regarding this point, I think TEMPO don't even have FloatingPositioner set in the current implementation. They are drawn as FloatingObjects similar to the dynamics or mordents. To leverage FloatingPositioner, first a bunch of code reordering would be needed and I'm not sure that's not going to break current functionality.

Reply:
Nvm my previous comment, I think I see where FloatingPositioner is set.